### PR TITLE
[FEATURE] Add Build Badge

### PR DESCRIPTION
**Pull request by TheFaR8**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [ ] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [X] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Added badge (for windows and linux)
thanks :)

### CODE REVIEW

**Comment by TheFaR8** [comment]
Will repair the PEP8 issue later (as I do this in school)

**Comment by TheFaR8** [comment]
PEP8 issue done

Now
I made them myself (even it looks like @MatejMecka code, it is not)

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
         db, kvm_name, TestPlatform.windows, repository, delay)
 
 
+@mod_ci.route('/ci_badge')
+def ci_badge():
+    dwg = svgwrite.Drawing()
+    commit = Test.query.filter(Test.test_type != Test.pull_request)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Did you test this? This code won't work at all.

File: mod_ci/controllers.py [hunk 2]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
...
+    release = Test.query.filter(Test.test_type != Test.commit)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Same here

File: mod_ci/controllers.py [hunk 3]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
...
+    if kvm_test.name is kvm_processor_linux:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Where does kvm_test come from? Where does kvm_processor_linux come from?

File: mod_ci/controllers.py [hunk 4]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
...
+        if commit.TestStatus = TestStatus.completed and Status.SUCCESS:
+            dwg.add(dwg.image(
+                    "https://img.shields.io/badge/Test-passed-brightgreen.svg",
+                    size=(4, 12))
+                    dwg.save()
+                    elif commit.TestStatus=TestStatus.canceled and Status.ERROR:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Your indendation is all off.

File: mod_ci/controllers.py [hunk 5]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
...
 
+@mod_ci.route('/ci_badge')
+def ci_badge():
+    dwg = svgwrite.Drawing()
+    commit = Test.query.filter(Test.test_type != Test.pull_request)
+    release = Test.query.filter(Test.test_type != Test.commit)
+    if kvm_test.name is kvm_processor_linux:
+        if commit.TestStatus = TestStatus.completed and Status.SUCCESS:
+            dwg.add(dwg.image(
+                    "https://img.shields.io/badge/Test-passed-brightgreen.svg",
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: We don't want do depend on external services to get an SVG image.

File: mod_ci/controllers.py [hunk 6]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
...
+                    size=(4, 12))
+                    dwg.save()
+                    elif commit.TestStatus=TestStatus.canceled and Status.ERROR:
+                    dwg.add(dwg.image(
+                        "https://img.shields.io/badge/Test-canceled-lightgrey.svg",
+                        size=(4, 12))
+                    dwg.save()
+                    return
+                    elif commit.TestStatus=TestStatus.completed and Status.FAILURE:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: All these if's make no sense. Determining the status of a test result is more complicated than this.

Besides, where is Status coming from?

File: mod_ci/controllers.py [hunk 7]
```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
         db, kvm_name, TestPlatform.windows, repository, delay)
 
 
+@mod_ci.route('/ci_badge')
+def ci_badge():
+    dwg = svgwrite.Drawing()
+    commit = Test.query.filter(Test.test_type != Test.pull_request)
+    release = Test.query.filter(Test.test_type != Test.commit)
+    if kvm_test.name is kvm_processor_linux:
+        if commit.TestStatus = TestStatus.completed and Status.SUCCESS:
```
Comment (MatejMecka) [inline mod_ci/controllers.py]: Also this is assigment not comparasion.

**Comment by TheFaR8** [comment]
can i do this task without svgwrite (open the svgs from external folder)
or must i do this task via svgwrite?

**Comment by canihavesomecoffee** [comment]
@TheFaR8 It all depends on what road you want to take. There's several ways to accomplish the task, and not all of them require svgwrite.

**Comment by saurabhshri** [comment]
I just commented on the issue : https://github.com/CCExtractor/sample-platform/issues/111#issuecomment-355594495

File: mod_ci/controllers.py [hunk 8]
```diff
@@ -442,6 +442,24 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
             platform=platform, id=test.id))
 
 
+def ci_badge(state):
+    if test_type != TestType.pull_request:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This won't work.

File: mod_ci/controllers.py [hunk 9]
```diff
@@ -442,6 +442,24 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+        if state == Status.SUCCESS:
+            with open('../static/badges/Test-Passed-brightgreen.svg',
+                      'r') as svg:
+                return svg.read()
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: What's the point of doing a return if you don't do anything with that return?

File: static/badges/Test-Canceled-lightgrey.svg [hunk 10]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="92" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="92" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h33v20H0z"/><path fill="#9f9f9f" d="M33 0h59v20H33z"/><path fill="url(#b)" d="M0 0h92v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"><text x="175" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="230">Test</text><text x="175" y="140" transform="scale(.1)" textLength="230">Test</text><text x="615" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="490">Canceled</text><text x="615" y="140" transform="scale(.1)" textLength="490">Canceled</text></g> </svg>
```
Comment (canihavesomecoffee) [inline static/badges/Test-Canceled-lightgrey.svg]: Did you create these images yourself?

File: mod_ci/controllers.py (hunk 9 above)
Comment (TheFaR8) [reply mod_ci/controllers.py]: i guess i have to change read into something else

File: static/badges/Test-Canceled-lightgrey.svg (hunk 10 above)
Comment (TheFaR8) [reply static/badges/Test-Canceled-lightgrey.svg]: I generated these images in shields.io and then i save them in my local (as a svg)

File: mod_ci/controllers.py (hunk 8 above)
Comment (TheFaR8) [reply mod_ci/controllers.py]: okay
will find another way to figure them out

File: static/badges/Test-Canceled-lightgrey.svg (hunk 10 above)
Comment (canihavesomecoffee) [reply static/badges/Test-Canceled-lightgrey.svg]: You're lucky that shields.io doesn't claim any copyright on the generated badges.

File: mod_ci/controllers.py (hunk 8 above)
Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: I'm talking just about this. test_type is undefined in this context.

Comment (TheFaR8) [reply mod_ci/controllers.py]: the right one is test.test_type (watching from the code below(other function))

Comment (TheFaR8) [reply mod_ci/controllers.py]: ohh
i must add test_type as a param

File: mod_ci/controllers.py [hunk 11]
```diff
@@ -442,6 +442,50 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
 
 
+def ci_badge_windows(status, platform, test_type, branch="master"):
+    platform = test.platform
+    if platform is TestPlatform.linux:
+        if test_type == TestType.commit:
+            if state == Status.SUCCESS:
+                with open('../static/badges/Windows_success.svg',
+                          'r') as svg:
+                    print svg.read()
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Printing it is just as useless as returning it. I'm unsure why you're doing that?

File: mod_ci/controllers.py [hunk 12]
```diff
@@ -442,6 +442,50 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
             platform=platform, id=test.id))
 
 
+def ci_badge_windows(status, platform, test_type, branch="master"):
+    platform = test.platform
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why pass in the platform if you immediately overwrite it?

File: mod_ci/controllers.py [hunk 13]
```diff
@@ -442,6 +442,50 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+    if platform is TestPlatform.linux:
+        if test_type == TestType.commit:
+            if state == Status.SUCCESS:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This variable is undefined.

File: mod_ci/controllers.py [hunk 14]
```diff
@@ -442,6 +442,50 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+                with open('../static/badges/Windows_success.svg',
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why split this line? We're nowhere close to 120 characters.

File: mod_ci/controllers.py [hunk 15]
```diff
@@ -442,6 +442,50 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+                          'r') as svg:
+                    print svg.read()
+        else:
+            return
+    else:
+        return
+
+
+def ci_badge_linux(status, platform, test_type, branch="master"):
+    platform = test.platform
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Same remarks apply here.

Besides, why duplicate this code just for a different badge name?

File: mod_ci/controllers.py [hunk 16]
```diff
@@ -749,6 +793,8 @@ def progress_reporter(test_id, token):
                 if status == TestStatus.canceled:
                     state = Status.ERROR
                     message = 'Tests aborted due to an error; please check'
+                    ci_badge_linux(state, linux, commit, master)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: the linux variable is undefined.

File: mod_ci/controllers.py [hunk 17]
```diff
@@ -749,6 +793,8 @@ def progress_reporter(test_id, token):
...
+                    ci_badge_windows(state, windows, commit, master)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Same here

**Comment by canihavesomecoffee** [comment]
Please make use of an decent IDE that can point out unused/undefined variables.

**Comment by TheFaR8** [comment]
my PC just too slow to run any IDE
sry

File: mod_ci/controllers.py (hunk 11 above)
Comment (TheFaR8) [reply mod_ci/controllers.py]: so I don't need them?
just read them and then leave right?

Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: And what does reading them achieve? They need to be displayable if you access a fixed url for that platform (see Saurabh's comment from yesterday)

Comment (TheFaR8) [reply mod_ci/controllers.py]: reading them only makes them well .. read
i am confused what to use to display them

Comment (TheFaR8) [reply mod_ci/controllers.py]: .show()?

Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: This is going nowhere like this. Please take some time to read the comment @saurabhshri  posted yesterday, and try to derive a solution from that comment.

Comment (TheFaR8) [reply mod_ci/controllers.py]: oh shutil

File: mod_ci/controllers.py [hunk 18]
```diff
@@ -442,6 +442,33 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+                        sampleplatform.ccextractor.org/buildtest/results)
+
+        else:
+            if status == Status.SUCCESS:
+                svg = open('../static/badges/Windows_success.svg')
+            elif status == Status.FAILURE:
+                svg = open('../static/badges/Windows_failed.svg')
+            else:
+                svg = open('../static/badges/Windows_error.svg')
+            svg.shutil(https://github.com/CCExtractor/sample-platform/static/badges,
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This makes zero sense. Please stop adding commits until you are certain that it all would work, and will yield *no* programming errors.

Comment (TheFaR8) [reply mod_ci/controllers.py]: okay

**Comment by saurabhshri** [comment]
Hey @TheFaR8 , thanks for the work but unfortunately this doesn't seem to be going in the right direction right now. Moreover an alternate PR for the same feature is already merged. Feel free to make any improvements to that! 

@canihavesomecoffee this can be closed.

**Comment by saurabhshri** [comment]
@TheFaR8 Looks like the current merged implementation can use some good SVG badges ;)

**Comment by TheFaR8** [comment]
It's okay
Thanks for the guidance 
Sorry for being retarded (not testing the code before doing a commit)

**Pull request closed without merging**